### ISSUE TITLE
Fragment signature and fragment local variables mechanism

### ISSUE THREAD

**Issue by danielfernandez** [thymeleaf/thymeleaf#181]
In order to create a more _function-like_ mechanism for the use of template fragments, fragments defined with `th:fragment` could be allowed to specify a sort of _parameters_:

``` html
<div th:fragment="frag (onevar,twovar)">
  <p th:text="${onevar} + ' - ' + ${twovar}">...</p>
</div>
```

This would require the use of one of these two syntaxes to call the fragment:

``` html
<div th:include="::frag (${value1},${value2})">...</div>
...
<div th:include="::frag (onevar=${value1},twovar=${value2})">...</div>
```

Note that order is not important in the last option:

``` html
<div th:include="::frag (twovar=${value2},onevar=${value1})">...</div>
```

**Fragment local variables without fragment signature**

Even if fragments are defined without signature, like this:

``` html
<div th:fragment="frag">
  ...
</div>
```

We could use the second syntax specified above to call them (and only the second one):

``` html
<div th:include="::frag (onevar=${value1},twovar=${value2})">
```

This would be, in fact, equivalent to a combination of `th:include` and `th:with`:

``` html
<div th:include="::frag" th:with="onevar=${value1},twovar=${value2}">
```

But the difference is, this will also work with `th:substituteby`/`th:replace`. Until now these attributes didn't work well with `th:with` because they effectively removed the host tag before `th:with` got executed:

``` html
<!-- th:with will never be executed. Variables will never be created -->
<div th:replace="::frag" th:with="onevar=${value1},twovar=${value2}">
```

**Context composition for fragments**

Note that this specification of local variables for a fragment --no matter whether it has a signature or not-- does not cause an initialization of the context to zero. Fragments will still be able to access every context variable being used at the calling template like they currently are.

This solves #87

**Comment by tadas-subonis** [comment]
I don't think you want to have such complex "th:with" style. JSP has solved these problems in a lot more fluent style. See http://docs.oracle.com/cd/E19502-01/819-3669/bnalv/index.html

So in this case it would be most useful to have:

``` xml
<div th:include="::identifier">
  <th:attr name="twovar" value="${value2}" />
  <th:attr name="onevar" value="${value1}" />
   <div>
     actual body that gets passed to parent tag
   </div>
</div>
```

**Comment by santoshballary** [comment]
above whatever u explaining is about different fragment from same page.
I need to include different fragments from different pages, based on the button click i need to replace old fragment to new fragment.

please help me on this

**Comment by danielfernandez** [comment]
@santoshballary maybe you should post a question explaining your scenario at the forum? http://forum.thymeleaf.org

**Comment by danielfernandez** [comment]
This feature has now been implemented and is available in 2.1.0-SNAPSHOT.

**Comment by danielfernandez** [comment]
Some associated tests: https://github.com/thymeleaf/thymeleaf-tests/blob/20130728/src/test-21/resources/engine21/attrprocessors/include/ tests 05 to 15.

**Comment by danielfernandez** [comment]
Version 2.1.0-m1 has been published as a milestone towards 2.1.0, including this feature. This way users who want to test this do not have to create a dependency on a snapshot version.
